Source/Xamarin/Prism.Forms/Navigation/Xaml/Navigation.cs

```diff
@@ -0,0 +1,132 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Windows.Input;
+using Prism.Common;
+using Prism.Ioc;
+using Xamarin.Forms;
+using Xamarin.Forms.Xaml;
+
+namespace Prism.Navigation.Xaml
+{
+    public abstract class Navigation : IMarkupExtension, ICommand
+    {
+        protected internal static readonly BindableProperty RaiseCanExecuteChangedInternalProperty =
+            BindableProperty.CreateAttached("RaiseCanExecuteChangedInternal",
+                typeof(Action),
+                typeof(Navigation),
+                default(Action));
+
+        public static readonly BindableProperty CanNavigateProperty =
+            BindableProperty.CreateAttached("CanNavigate",
+                typeof(bool),
+                typeof(Navigation),
+                true,
+                propertyChanged: OnCanNavigatePropertyChanged);
+
+        protected internal static readonly BindableProperty NavigationServiceProperty =
+            BindableProperty.CreateAttached("NavigationService",
+                typeof(INavigationService),
+                typeof(Navigation),
+                default(INavigationService));
+
+
+        protected BindableObject Bindable;
+        protected IEnumerable<BindableObject> BindableTree;
+        protected bool IsNavigating;
+
+        public bool AllowDoubleTap { get; set; } = false;
+
+        public Page Page { protected get; set; }
```

Comment:
I am thinking this should be called SourcePage to imply which page it rally is. I can see "Page" being a little too ambiguous

Reply:
Done